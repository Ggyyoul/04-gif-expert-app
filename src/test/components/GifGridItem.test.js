import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem/GifGridItem';

describe('Tests in component <GifGridItem />', () => {

	const title = 'A Title';
	const url = 'https://localhost/holix.jpg'
	let wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should mount the component properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

	test('should have a p with de title', () => {
		const p = wrapper.find('p')
		expect(p.text().trim()).toBe(title)	
	});

	test('should the image have a url and alt same that props', () => {
		const img = wrapper.find('img')
		// console.log(img.html()) img.prop('src')
		
		expect(img.prop('src')).toBe(url)
		expect(img.prop('alt')).toBe(title)
	});

	test('should have animate__bounce', () => {
		const div = wrapper.find('div')
		//forma personal de validad la clase
		expect(div.hasClass('animate__bounce')).toBe(true)
		//forma del video de validar la clase
		const className= div.prop('className')
		expect(className.includes('animate__bounce')).toBe(true)
	})
	
	
	
});

