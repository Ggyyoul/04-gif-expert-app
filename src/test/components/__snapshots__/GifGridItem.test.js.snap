// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in component <GifGridItem /> should mount the component properly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="A Title"
    src="https://localhost/holix.jpg"
  />
  <p>
    A Title
  </p>
</div>
`;
